src/types/json.h

```diff
@@ -106,6 +108,44 @@ struct JsonValue {
     return Status::OK();
   }
 
+  Status StrAppend(std::string_view path, const std::string &append_value, std::vector<int64_t> &append_cnt) {
+    try {
+      auto append_trimmed = util::Trim(append_value, "\"");
+      jsoncons::jsonpath::json_replace(
+          value, path, [&append_trimmed, &append_cnt](const std::string & /*path*/, jsoncons::json &origin) {
+            if (origin.is_string()) {
+              auto origin_trimmed = origin.as<std::string>();
+              append_cnt.push_back((int64_t)(origin_trimmed.length() + append_trimmed.length()));
+              std::string new_value;
+              new_value.append("\"").append(origin_trimmed).append(append_trimmed).append("\"");
+              origin = jsoncons::json::parse(new_value);
+            } else {
+              append_cnt.push_back(-1);
+            }
+          });
```

Comment:
Same as https://github.com/apache/kvrocks/pull/1841#discussion_r1367656620.

And I think it will be a **blocking issue** of this PR.